linker/shared/src/main/scala/org/scalajs/linker/backend/BasicLinkerBackend.scala

```diff
@@ -120,6 +132,13 @@ final class BasicLinkerBackend(config: LinkerBackendImpl.Config)
 
 private object BasicLinkerBackend {
   private final class PrintedModuleSetCache(withSourceMaps: Boolean) {
+    private var lastHeader: String = null
+    private var lastFooter: String = null
+
+    private var _headerBytesCache: Array[Byte] = null
+    private var _footerBytesCache: Array[Byte] = null
+    private var _headerNewLineCountCache: Int = 0
```

Comment:
The header and the footer are part of the config. They cannot change between incremental runs. Do this conversion once when creating the `Emitter`?

Reply:
The header can contain top level exports for the `NoModule` kind. So that wouldn't work.